### PR TITLE
feat(multi-language) #30591 : Create endpoints to feed the Multi-language Component for the Edit Contentlet sidebar

### LINKED ISSUE #30591: [Multi-language] : Create endpoints to feed the Multi-language Component for the Edit Contentlet sidebar
### Parent Issue

* #30581

### User Story

As a dotCMS User, I want to view locale-related information of the Contentlet that I'm creating or editing.

### Acceptance Criteria

Expose the required data via REST:

* An Endpoint that takes a Contentlet Identifier which returns ALL languages in the dotCMS repo. For each of them, a new attribute will indicate whether the Contentlet is available in such a language or not.
* An Endpoint that returns the currently selected default Language.
* The data structure will be the same one used by the soon-to-be-deprecated method in the `PageResource` class. For instance:
```json
{
    "entity": [
        {
            "country": "United States",
            "countryCode": "US",
            "id": 1,
            "isoCode": "en-us",
            "language": "English",
            "languageCode": "en",
            "translated": true
        },
        {
            "country": "Espana",
            "countryCode": "ES",
            "id": 2,
            "isoCode": "es-es",
            "language": "Espanol",
            "languageCode": "es",
            "translated": false
        }
    ],
    "errors": [],
    "i18nMessagesMap": {},
    "messages": [],
    "pagination": null,
    "permissions": []
}
```

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.slack.com/archives/C058GCQ7T39/p1730827058966279

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

* [ ] Implement the missing features as described in my comment in the spike ticket: https://github.com/dotCMS/core/issues/30581#issuecomment-2460430054
* [ ] Add the respective Postman tests.

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
### Proposed Changes
* A new Endpoint has been added to retrieve the languages that a contentlet is available on: `/api/v1/content//{identifier}/languages`
* A new Endpoint has been added to retrieve the default language: `/api/v2/languages/_getdefault`
* The existing Endpoint that retrieved the languages that an HTML Page is available on has been marked as `Deprecated` -- as well as its related code -- in order to favor the new Endpoint.

This PR fixes: #30591

### CODE REVIEW

**Review by freddyDOTCMS — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentResource.java [hunk 1]
```diff
@@ -590,5 +594,70 @@ public Response pullRelated(@Context final HttpServletRequest request,
...
+     * whether the Contentlet is available in such a language or not.
+     *
+     * @throws DotDataException An error occurred when interacting with the database.
+     */
+    @GET
+    @Path("/{identifier}/languages")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response checkContentLanguageVersions(@Context final HttpServletRequest request,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentResource.java:622]: Avoid Response, use a ReponseEntityView

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 2]
```diff
@@ -637,4 +637,26 @@ public Response getIsoLanguagesAndCountries (
...
+     * instance.
+     *
+     * @throws DotDataException An error occurred when retrieving the default {@link Language}.
+     */
+    @GET
+    @Path("/_getdefault")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response getDefaultLanguage(@Context final HttpServletRequest request,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:656]: Same thing, use a ResponseEntityVIew

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentResource.java (hunk 1 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentResource.java]: Done!

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java (hunk 2 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Done!

**Pull request merged**